src/uct/ib/rc/accel/gga_mlx5.c

```diff
@@ -205,48 +442,93 @@ static uct_iface_ops_t uct_gga_mlx5_iface_tl_ops = {
+    .ep_get_address           = uct_gga_mlx5_ep_get_address,
     .ep_connect_to_ep         = ucs_empty_function_return_unsupported,
-    .iface_flush              = ucs_empty_function_return_unsupported,
-    .iface_fence              = (uct_iface_fence_func_t)ucs_empty_function_do_assert,
-    .iface_progress_enable    = ucs_empty_function_do_assert_void,
-    .iface_progress_disable   = ucs_empty_function_do_assert_void,
-    .iface_progress           = (uct_iface_progress_func_t)ucs_empty_function_do_assert,
-    .iface_event_fd_get       = ucs_empty_function_return_unsupported,
-    .iface_event_arm          = ucs_empty_function_return_unsupported,
+    .iface_flush              = uct_rc_iface_flush,
+    .iface_fence              = uct_rc_iface_fence,
+    .iface_progress_enable    = uct_base_iface_progress_enable,
+    .iface_progress_disable   = uct_base_iface_progress_disable,
+    .iface_progress           = uct_gga_mlx5_iface_progress,
+    .iface_event_fd_get       = uct_gga_mlx5_iface_event_fd_get,
+    .iface_event_arm          = uct_gga_mlx5_iface_arm,
     .iface_close              = uct_gga_mlx5_iface_t_delete,
-    .iface_query              = (uct_iface_query_func_t)ucs_empty_function_do_assert,
+    .iface_query              = uct_gga_mlx5_iface_query,
     .iface_get_address        = ucs_empty_function_return_success,
-    .iface_get_device_address = ucs_empty_function_return_unsupported,
+    .iface_get_device_address = uct_gga_mlx5_iface_get_device_address,
     .iface_is_reachable       = uct_base_iface_is_reachable
 };
 
+static int
+uct_gga_mlx5_iface_is_reachable_v2(const uct_iface_h tl_iface,
+                                   const uct_iface_is_reachable_params_t *params)
+{
+    uct_ib_iface_t *iface   = ucs_derived_of(tl_iface, uct_ib_iface_t);
+    uct_ib_device_t *device = uct_ib_iface_device(iface);
+    const uct_gga_mlx5_dev_addr_t *dev_addr;
+    uct_iface_reachability_scope_t scope;
+
+    if (!uct_iface_is_reachable_params_addrs_valid(params)) {
+        return 0;
+    }
+
+    dev_addr = (const uct_gga_mlx5_dev_addr_t*)params->device_addr;
+    if (dev_addr->be_sys_image_guid !=
+        device->dev_attr.orig_attr.sys_image_guid) {
+        return 0;
+    }
+
+    scope = UCS_PARAM_VALUE(UCT_IFACE_IS_REACHABLE_FIELD, params, scope, SCOPE,
+                            UCT_IFACE_REACHABILITY_SCOPE_NETWORK);
+    return uct_ib_iface_dev_addr_is_reachable(iface, &dev_addr->ib_addr, scope);
+}
+
+static ucs_status_t
+uct_gga_mlx5_iface_init_rx(uct_rc_iface_t *rc_iface,
+                           const uct_rc_iface_common_config_t *rc_config)
+{
+    uct_rc_mlx5_iface_common_t *iface =
+            ucs_derived_of(rc_iface, uct_rc_mlx5_iface_common_t);
+
+    iface->rx.srq.type    = UCT_IB_MLX5_OBJ_TYPE_NULL;
+    iface->rx.srq.srq_num = 0;
+    return UCS_OK;
+}
+
+static void uct_gga_mlx5_iface_qp_cleanup(uct_rc_iface_qp_cleanup_ctx_t *ctx)
```

Comment:
I don't think we need cleanup mechanism for GGA. Cleanup is about waiting for LAST_WQE_REACHED event, which is not relevant since we don't have SEND/RECV operations. I'd suggest to not add it in this PR, unless we see some failure. Can test error handling and add it later if needed.

Reply:
I avoid `LAST_WQE_REACHED` flow adding `async` parameter which is false for GGA, this approach adds some overhead on `ep_destroy` but re-uses dtors of base classed without significant change, otherwise need to refactor them deeper